src/connector/src/parser/mod.rs

```diff
@@ -657,7 +677,17 @@ async fn into_chunk_stream<P: ByteStreamSourceParser>(mut parser: P, data_stream
         let process_time_ms = chrono::Utc::now().timestamp_millis();
         for (i, msg) in batch.into_iter().enumerate() {
             if msg.key.is_none() && msg.payload.is_none() {
-                tracing::debug!(offset = msg.offset, "skip parsing of heartbeat message");
+                tracing::debug!(
+                    offset = msg.offset,
+                    "got a empty message, could be a heartbeat"
+                );
+                let mut writer = builder.row_writer().with_meta(MessageMeta {
+                    meta: &msg.meta,
+                    split_id: &msg.split_id,
+                    offset: &msg.offset,
+                });
+                writer.record_empty_row_idx(i);
```

Comment:
This is not correct when parsing with cdc transaction, which would parse multiple `batch` before emitting a chunk.